adafruit_logging.py

```diff
@@ -296,23 +307,47 @@ def getEffectiveLevel(self) -> int:
         return self._level
 
     def addHandler(self, hdlr: Handler) -> None:
-        """Sets the handler of this logger to the specified handler.
-
-        *NOTE* This is slightly different from the CPython equivalent
-        which adds the handler rather than replacing it.
+        """Adds the handler to this logger.
 
         :param Handler hdlr: The handler to add
         """
-        self._handler = hdlr
+        self._handlers.append(hdlr)
+
+    def removeHandler(self, hdlr: Handler) -> None:
+        """Remove handler from this logger.
+
+        :param Handler hdlr: The handler to remove
+        """
+        self._handlers.remove(hdlr)
 
     def hasHandlers(self) -> bool:
         """Whether any handlers have been set for this logger"""
-        return self._handler is not None
+        return len(self._handlers) > 0
 
     def _log(self, level: int, msg: str, *args) -> None:
         record = _logRecordFactory(self.name, level, msg % args, args)
-        if self._handler and level >= self._level:
-            self._handler.emit(record)
+        self.handle(record)
+
+    def handle(self, record: LogRecord) -> None:
+        """Pass the record to all handlers registered with this logger.
+
+        :param LogRecord record: log record
+        """
+        if not self.hasHandlers() and not self.emittedNoHandlerWarning:
+            sys.stderr.write(f"Logger '{self.name}' has no handlers\n")
+            self.emittedNoHandlerWarning = True
+            return
+
+        if record.levelno >= self._level:
+            for handler in self._handlers:
+                if record.levelno >= handler.level:
+                    try:
+                        handler.emit(record)
+                    except Exception as e:  # pylint: disable=broad-except
+                        if sys.stderr and printHandlerExceptions:
+                            sys.stderr.write(
+                                f"Handler {handler} produced exception: {e}\n"
+                            )
```

Comment:
I think it's okay to let this fail rather than try to handle failures.  I agree that there's a risk a single handler won't work, but that it's okay to leave it up to the user to verify it will.  An example of where this might matter is where logging needs to work and it isn't hooked up to a serial monitor of any kind.  In this case, the handler will catch the failure and write and continue to operate , but will seem functional otherwise.  This also gives the user a chance to check for more specific exceptions.

Reply:
Fine with me ! Esp. the wide catch seems dicey.